File: 0918-maximum-sum-circular-subarray/0918-maximum-sum-circular-subarray.py
class Solution:
    def maxSubarraySumCircular(self, nums: List[int]) -> int:
        total = cur_max = cur_min = 0
        max_sum = min_sum = nums[0]
        for num in nums:
            cur_max = max(num, num + cur_max)
            max_sum = max(cur_max, max_sum)
            cur_min = min(num, num + cur_min)
            min_sum = min(cur_min, min_sum)
            total += num
        return max(max_sum, total - min_sum) if max_sum > 0 else max_sum

File: 0918-maximum-sum-circular-subarray/0918-maximum-sum-circular-subarray.cs
public class Solution {
    public int MaxSubarraySumCircular(int[] nums) {
        int total, curMax, curMin, maxSum, minSum;
        total = curMax = curMin = 0;
        maxSum = minSum = nums[0];
        foreach (var num in nums){
            curMax = Math.Max(num, num + curMax);
            maxSum = Math.Max(curMax, maxSum);
            curMin = Math.Min(num, num + curMin);
            minSum = Math.Min(curMin, minSum);
            total += num;
        }
        return maxSum > 0 ? Math.Max(maxSum, total - minSum) : maxSum;
    }
}